src/promiseQueue.ts

```diff
@@ -0,0 +1,43 @@
+/*
```

Comment:
overall comment...there's some tools that do this kind of thing if we find a lot more use cases for it.

https://github.com/supercharge/promise-pool is fairly small, zero-dep.  The error handling and abort looks nice.